torchtune/utils/metric_logging.py

```diff
@@ -317,3 +317,127 @@ def close(self) -> None:
         if self._writer:
             self._writer.close()
             self._writer = None
+
+
+class CometLogger(MetricLoggerInterface):
+    """Logger for use w/ Comet (https://www.comet.com/site/).
+    For more information about arguments expected by Comet, see
+    https://www.comet.com/docs/v2/guides/experiment-management/configure-sdk/#for-the-experiment.
+
+    Args:
+        api_key (Optional[str]): Comet API key. It's recommended to configure the API Key from the environment.
```

Comment:
should we just encourage users to use `comet login`?

Reply:
Yes we should encourage users to use `comet login`.

Saving your Comet API Key in a git tracked file is a bad practice and we are of course not recommending it. Having the possibility to set your Comet API Key in your configuration is useful when you are rendering your configuration file and injecting the Comet API Key automatically from a secret manager place.

I do not know if that's something that's frequent when using torchtune.

Reply:
Actually I expect people will mostly just use the API key? By exporting it before running `tune run ...`. Otherwise, where would they call `comet.login()`?